rero_ils/modules/api.py

```diff
@@ -681,10 +690,23 @@ def _prepare_record(record, index, doc_type, arguments=None, **kwargs):
         :param **kwargs: Extra parameters.
         :return: The record metadata.
         """
-        if current_app.config['INDEXER_REPLACE_REFS']:
-            data = record.replace_refs().dumps()
+        if not getattr(record, 'enable_jsonref', True):
+            # If dumper is None, dumps() will use the default configured dumper
+            # on the Record class.
+            data = record.dumps(dumper=self.record_dumper)
         else:
-            data = record.dumps()
+            # Old-style dumping - the old style will still if
+            # INDEXER_REPLACE_REFS is False use the Record.dumps(), however the
+            # default implementation is backward compatible for new-style
+            # records. Also, we're adding extra information into the record
+            # like _created and _updated afterwards, which the Record.dumps()
+            # have no control over.
+            if current_app.config['INDEXER_REPLACE_REFS']:
```

Comment:
It could be safer to use `current_app.config.get('INDEXER_REPLACE_REFS', False|True)`